main.go

```diff
@@ -84,63 +86,62 @@ func main() {
 }
 
 func run(cfg Config, additionalSchemaPaths []string, updateDependencies bool) error {
-	return foreachChart(cfg.ChartsDirectory.path, func(base string) error {
-		logger := log.With().Str("chart", filepath.Base(base)).Logger()
-		valuesFiles, err := os.ReadDir(filepath.Join(base, TestsPath))
+	var validations_errors []error
 
+	err := filepath.WalkDir(cfg.ChartsDirectory.path, func(path string, dirent fs.DirEntry, err error) error {
+		logger := log.With().Str("path", path).Logger()
 		if err != nil {
```

Comment:
Could we turn this into a series of early returns to reduce the rightward drift? Something like:

```go
if err != nil {
  logger.Warn().Err(err).Msg("skipping path (error)")
  return nil
}

if dirent.isDir() && skipRegex.MatchString(dirent.Name()) {
  logger.Info().Msg(…)
  return nil
}

if dirent.Name() != "Chart.yaml" && dirent.Name() != "Chart.yml" {
  return nil
}

chartDir := filepath.Dir(…)
// …
return fs.SkipDir
```